File: programmers/코딩테스트연습/해쉬/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool solution(vector<string> phone_book) {
    sort(phone_book.begin(), phone_book.end());
    for(int i = 0; i < phone_book.size() - 1; i++){
        int size = phone_book[i].size();
        if(phone_book[i+1].substr(0, size) == phone_book[i]){
            return false;
        }
    }
    return true;
}

File: programmers/해쉬/전화번호 목록.py
def solution(phone_book):
    phone_book = sorted(phone_book)
    for i in range(len(phone_book)):
        if phone_book[i] == phone_book[i+1][:len(phone_book[i])]:
            return False
    return True

if __name__ == "__main__":
    print(solution(["119", "97674223", "1195524421"]))
    # print(solution(["123", "456", "789"]))
